### DISCUSSION TITLE
top sources and countries sections not working

### DISCUSSION THREAD

**Discussion by Janevski** [plausible/analytics#1920] [Self-Hosted Support]
I am using the :latest docker tag image of plausible and i don't know how to check for a version number.
I am not using maxmind, just db-ip as it comes by default.

Top Sources and Countries sections state no data yet, when there is data present.
Countries map also does not work.
![Screenshot from 2022-05-22 10-51-02](https://user-images.githubusercontent.com/5252703/169688724-47b06261-0878-4ad8-b0b0-e86739427ae5.png)

**Expected behavior**
Countries map should work.
Top Sources should show something as well.

- OS:Ubuntu 20.04 via docker image
- Browser: Chrome 
- Browser Version: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36

**Comment by EmberLightVFX** [comment]
I'm having the same problem. I do have data under Top Sources but the countries are empty.
I'm running my instance through docker/portainer and this is the code I'm using:

`
version: "3.3"
services:

  geoip:
    image: maxmindinc/geoipupdate
    restart: always
    environment:
      - GEOIPUPDATE_EDITION_IDS=GeoLite2-Country
      - GEOIPUPDATE_FREQUENCY=168 # update every 7 days
    env_file:
      - stack.env
    volumes:
      - /volume2/docker/plausible/geoip:/usr/share/GeoIP

  plausible_db:
    image: postgres:12
    restart: unless-stopped
    volumes:
      - /volume2/docker/plausible/db-data:/var/lib/postgresql/data
    environment:
      - POSTGRES_PASSWORD=postgres
    networks:
      - plausible

  plausible_events_db:
    image: yandex/clickhouse-server:21.3.2.5
    restart: unless-stopped
    volumes:
      - /volume2/docker/plausible/event-data:/var/lib/clickhouse
      - /volume2/docker/plausible/clickhouse/clickhouse-config.xml:/etc/clickhouse-server/config.d/logging.xml:ro
      - /volume2/docker/plausible/clickhouse/clickhouse-user-config.xml:/etc/clickhouse-server/users.d/logging.xml:ro
    ulimits:
      nofile:
        soft: 262144
        hard: 262144
    networks:
      - plausible

  plausible:
    image: plausible/analytics:latest
    restart: unless-stopped
    command: sh -c "sleep 10 && /entrypoint.sh db createdb && /entrypoint.sh db migrate && /entrypoint.sh db init-admin && /entrypoint.sh run"
    depends_on:
      - plausible_db
      - plausible_events_db
      - geoip
    volumes:
      - /volume1/docker/plausible/geoip:/geoip:ro
    environment:
      - GEOLITE2_COUNTRY_DB=/geoip/GeoLite2-Country.mmdb
    ports:
      - 8086:8000
    env_file:
      - stack.env
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.plausible.rule=Host(`ana.xxxxxxxx.xxxx`)"
      - "traefik.http.routers.plausible.entrypoints=web, websecure"
      - "traefik.http.routers.plausible.tls=true"
      - "traefik.http.routers.plausible.tls.certresolver=production"
      - "traefik.http.services.plausible.loadbalancer.server.port=8000"
      - "traefik.http.routers.plausible.middlewares=plausible@docker"
      - "traefik.http.middlewares.plausible.replacepathregex.regex=\\/js\\/ana\\.js"
      - "traefik.http.middlewares.plausible.replacepathregex.replacement=/js/plausible.js"
      - "traefik.docker.network=traefik"
    networks:
      - plausible
      - traefik

networks:
  traefik:
    external: true
  plausible:
    name: plausible
`

I don't understand why it's not wokring :/
db-ip.com shows all my info correct.

**Comment by ghost** [comment]
Probably the real IP of the visitors is not reaching plausible. Check the proxy.

**Comment by EmberLightVFX** [comment]
The problem was how traefik was connecting plausible through docker. I changed to caddy-gen and it now works as expected :)

**Comment by StrandedKitty** [comment]
For nginx the solution is to setup headers for your reverse proxy like this:
```
location / {
    proxy_set_header Host $host;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_pass http://whatever;
}
```

**Reply by Doublefire-Chen** [reply]
Thank you, bro. 👍👍👍Your answer which solved my same issue indeed help me a lot. I learned much from your answer.

**Reply by jkb-kt** [reply]
Having the same issue with the map data, using nginx reverse proxy but adding the lines above didn't help :( Not sure what else to check.

**Reply by CodeWithKyrian** [reply]
Thanks so much. This solved the problem for me
